modules/network-peering/README.md

```diff
@@ -0,0 +1,62 @@
+# Google Network Peering
+
+This module allows creation of a [VPC Network Peering](https://cloud.google.com/vpc/docs/vpc-peering) between two networks.
+
+The resources created/managed by this module are:
+
+- one network peering from `local network` to `peer network`
+- one network peering from `peer network` to `local network`
+
+## Usage
+
+Basic usage of this module is as follows:
+
+```hcl
+module "peering" {
+  source = "terraform-google-modules/terraform-google-network/google//modules/network-peering"
+
+  prefix        = "name-prefix"
+  local_network = "<FIRST NETWORK SELF LINK>"
+  peer_network  = "<SECOND NETWORK SELF LINK>"
+}
+```
+
+If you need to create more than one peering for the same VPC Network `(A -> B, A -> C)` you have to use output from the first module as an input to the second one in order to keep order of peering creation (It is not currently possible to create more than one peering connection for a VPC Network at the same time).
```

Comment:
One alternative pattern to consider is the pattern that @ideasculptor suggested [here](https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/pull/310#discussion_r343297443) and add a simple `complete` output to this module and a `depends_on` input which could be used for arbitrary chaining. We can assess in a follow-up though. @aaron-lane @ludoo for thoughts.

Reply:
Intriguing idea, though I'd be a bit wary of the additional complexity, and the need of unraveling how it works which might be beyond some users.

Reply:
Yeah, we can explore it in a follow-up. Ideally end users shouldn't have to unravel it at all: they just need to pass in:

`depends on = module1.complete`.

Reply:
> One alternative pattern to consider is the pattern that @ideasculptor suggested [here](https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/pull/310#discussion_r343297443) and add a simple `complete` output to this module and a `depends_on` input which could be used for arbitrary chaining. We can assess in a follow-up though. @aaron-lane @ludoo for thoughts.
That's interesting, only one thing, assuming that variable `depends_on` is reserved by terraform itself which alternative name would you recommend?
I think `depend_on` would be ok, WDYT?

Reply:
I don't think `depends_on` is reserved by Terraform actually, since it's not implemented on modules.

Reply:
> I don't think `depends_on` is reserved by Terraform actually, since it's not implemented on modules.

But it is https://www.terraform.io/upgrade-guides/0-12.html#reserved-variable-names
Suppose it's reserved since they want to support it in the feature releases

Reply:
Got it. `depend_on` would work then.

Reply:
@ludoo @morgante please take a look when you have time, that makes code a bit more complex, but the usage is simpler when you need a few peerings. 
I also updated the example to cover multi peering case with tests.

Reply:
> Got it. `depend_on` would work then.

I think this name `module_depends_on` a  bit more clear. Found similar example here https://discuss.hashicorp.com/t/tips-howto-implement-module-depends-on-emulation/2305